### ISSUE TITLE
CDATA[UnknownError: invalid code lengths set while uploading file in remote server

### ISSUE THREAD

**Issue by Ravi-ratnakaram** [SeleniumHQ/selenium#640]
Hi All,

I am using selenium webdriver version 2.45.1 and upload functionality is failing with error code "failure message="invalid code lengths set"><![CDATA[UnknownError: invalid code lengths set".

code snippet:
    nemo.driver.setFileDetector(new remote.FileDetector());
     var imgPath=path.join(__dirname,'..','test',img.jpg)
            driver.element.sendKeys(imgPath);

Could someone suggest the fix? or workaround for this issue.

Make a note, issue occurs only while running in remote and local runs it's perfectly alright.

Thanks for your help in advance...

**Comment by raghavan85** [comment]
I am facing the same issue as well.

**Comment by lukeis** [comment]
please provide a complete reproducible test script, we can then reopen the issue. But right now this is not enough to reproduce.

**Comment by raghavan85** [comment]
Reproduced in protractor:

Message:
Failed: invalid code lengths set
Stack:
UnknownError: invalid code lengths set
at Array.forEach (native)
From: Task: WebElement.sendKeys()
at Array.forEach (native)
at Array.forEach (native)

code snippet:
var fileToUpload = ('Doc1.docx'),
absolutePath = path.resolve(__dirname, fileToUpload);

var remote = require('../../../node_modules/protractor/node_modules/selenium-webdriver/remote');
browser.setFileDetector(new remote.FileDetector);
element(by.id('fileupload')).sendKeys(absolutePath);

Please reopen the issue. Let me know if you are still unable to reproduce this issue. You will need to a remote selenium grid address to hit. Works fine if hitting the local grid or directly.

**Comment by lukeis** [comment]
you would need to supply the file you are uploading. I can't reproduce this scenario with python, but it could be a problem specifically with the selenium-webdriver JS bindings.

**Comment by raghavan85** [comment]
Steps:
1. Install node, protractor on your machine.
2. Code snippet:

describe("File upload test", function(){
	it("remote file upload", function(){
	  var fileToUpload = ('../../input/fileuploads/Doc1.docx'),
        absolutePath = path.resolve(__dirname, fileToUpload);
       var remote = require('../../../node_modules/protractor/node_modules/selenium-webdriver/remote');
        browser.setFileDetector(new remote.FileDetector);
        element(by.id('fileupload')).sendKeys(absolutePath);
	});
});
save as test.spec.js
3. Create an endpoint with an element to upload a file 'fileupload' id. 
4. Start your webdriver in a different host than where the test file and
5. Create conf.js as below:

exports.config = {
  specs: [
    '**/*.spec.js'
  ],
  seleniumAddress: 'http://some_different_host_B:4444/wd/hub',
  baseUrl: 'http://some_host_A/that_has_file_upload'
};

6. run the following: protractor conf.js to reproduce the error.

**Comment by jleyba** [comment]
What OS are you running on?

**Comment by raghavan85** [comment]
Mac OS.

**Comment by raghavan85** [comment]
Hi guys, is there any update on this issue ?

**Comment by Ravi-ratnakaram** [comment]
Can someone please look into this issue and update as soon as possible. appreciate your help.

**Comment by Mikrovolnovka** [comment]
Selenium Server's output
```sh
WARN - Exception thrown
java.util.zip.ZipException: invalid code lengths set
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
	at java.util.zip.ZipInputStream.read(ZipInputStream.java:194)
	at java.io.FilterInputStream.read(FilterInputStream.java:107)
	at org.openqa.selenium.io.Zip.unzipFile(Zip.java:173)
	at org.openqa.selenium.io.Zip.unzip(Zip.java:155)
	at org.openqa.selenium.io.Zip.unzip(Zip.java:126)
	at org.openqa.selenium.remote.server.handler.UploadFile.call(UploadFile.java:42)
	at org.openqa.selenium.remote.server.handler.UploadFile.call(UploadFile.java:1)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by jesucarr** [comment]
Same issue with the latest 2.47.1 version

**Comment by irrenhaus** [comment]
+1
May be related to https://github.com/cthackers/adm-zip/issues/133

**Comment by irrenhaus** [comment]
My tests did now run roughly 100 times, each time uploading 2 files. They did run without a single error.
I rewrote the remote.FileDetector to do two things: 1. to be able to handle HTTP links (download the file with the given URI, Zip it, upload it to Selenium) and 2. (which is the important part here) I rewrote it to use an other archiving library than Adm-Zip.
It can be found here and should be easy to adapt to your needs (just have a look at the original to find the differences): https://gist.github.com/irrenhaus/55fb5046278ba97556b3

**Comment by jleyba** [comment]
I can't reproduce this on (node v0.12.7 or v4.0.0). Maybe I'm uploading the wrong type of file? It would be great if someone could contribute a file to our regression suite that this will reproduce with.

@irrenhaus I can start working on replacing `adm-zip` with `archiver` if that fixes this. I'll have no way of know it _actually_ fixes it though since I can't reproduce it to begin with :)

**Comment by irrenhaus** [comment]
@jleyba Sadly I have no files left which definitely caused this issue. Since I needed to generate a new file each time I was uploading (application checks if a file was already uploaded using a checksum of the file), I don't even know if this issue is dependent on the file or not. However, the files I uploaded always were only PDF files (first generated using wkhtmltopdf, later generated using golang & jung-kurt/gofpdf).

I can confirm that using `archiver` fixes the issue for me since all my uploads went without any error since I'm using the code I linked above.
